store/root.js

```diff
@@ -1,44 +1,28 @@
 import { observable, action, computed } from 'mobx'
 
-import activities from './activities'
 import User from './user'
-import DepositDates from './deposit-dates'
-import Works from './works'
 
-import apiRequest from 'api'
-
-class Root {
-  @observable user = new User()
-
-  @observable dataProvider = null
-
-  @observable activity = null
-
-  @observable statistics = {
-    metadataCount: null,
-    fullTextCount: null,
-    doiCount: null,
-  }
-
-  @observable plugins = {
-    discovery: null,
-    recommender: null,
+class Application {
+  constructor(config) {
+    this.config = config
   }
 
-  @observable works = null
-
-  @observable depositDates = null
-
   @computed
   get dataProviders() {
     return this.user.dataProviders
   }
 
-  @action async init(dataProviderId, activityPath) {
-    await this.user.init()
+  @action
+  async init(dataProviderId) {
+    const { config } = this
+    try {
+      this.user = observable(new User())
+    } catch (unauthorisedUser) {
+      if (!config.allowAnonymous) throw unauthorisedUser
```

Comment:
What happened to this anonymous access? I found it very cool and worth of implementing.

Reply:
I wanted to do this but I took too much in this PR. I wanted to close this PR because we already have more or less similar things implemented.

Although, I reimplemented all the stuff #207 and did not touch this one.